### PR TITLE
Move ChecksumHandle to CRC32C

### PR DESCRIPTION
**Pull request by wenkokke**
Moves the `ChecksumHandle` type to `CRC32C` to enable reuse (without exporting it from `RunBuilder`).

### CODE REVIEW

File: src/Database/LSMTree/Internal/CRC32C.hs [hunk 1]
```diff
@@ -65,8 +74,13 @@ import           Data.Word
 import           GHC.Exts
 import qualified GHC.ForeignPtr as Foreign
 import           System.FS.API
+
```
Comment (jorisdral) [inline src/Database/LSMTree/Internal/CRC32C.hs:77]:
```suggestion
```

and run `stylish-haskell`

File: src/Database/LSMTree/Internal/CRC32C.hs [hunk 2]
```diff
@@ -349,3 +363,58 @@ formatChecksumsFile checksums =
           <> BS.char8 '\n'
         | (ChecksumsFileName name, CRC32C crc) <- Map.toList checksums ]
 
+{-------------------------------------------------------------------------------
+  ChecksumHandle
+-------------------------------------------------------------------------------}
+
+{-| $checksum-handles
+    A handle ('ChecksumHandle') that maintains a running CRC32 checksum.
+-}
```
Comment (jorisdral) [inline src/Database/LSMTree/Internal/CRC32C.hs:372]: I think you can leave out the `|`

File: src/Database/LSMTree/Internal/RunBuilder.hs [hunk 3]
```diff
@@ -288,7 +286,7 @@ copyBlob builder@RunBuilder {..} blobref = do
     blob <- BlobRef.readRawBlobRef runBuilderHasFS blobref
     writeBlob builder blob
     --TODO: can't easily switch this to use BlobFile.writeBlob because
-    -- RunBuilder currently does everything uniformly with ChecksumHandle.
+    -- RunBuilder currently does everything uniformly with CRC.
```
Comment (jorisdral) [inline src/Database/LSMTree/Internal/RunBuilder.hs:289]: Probably accidental
```suggestion
    -- RunBuilder currently does everything uniformly with ChecksumHandle.
```

**Review by jorisdral — APPROVED** [review]

**Comment by wenkokke** [comment]
Closing in favour of #471.

**Pull request closed without merging**